google/cloud/bigquery/job/query.py

```diff
@@ -198,6 +198,39 @@ def from_api_repr(cls, stats: Dict[str, str]) -> "DmlStats":
         return cls(*args)
 
 
+class SearchReason(typing.NamedTuple):
+    """docstring"""
+
+    code: str
+    message: str
+    baseTable: TableReference
+    indexName: str
+
+    @classmethod
+    def from_api_repr(cls, reason):
+        code = reason.get("code")
+        message = reason.get("message")
+        baseTable = reason.get("baseTable")
+        indexName = reason.get("indexName")
+
+        return cls(code, message, baseTable, indexName)
+
+
+class SearchStats(typing.NamedTuple):
+    """docstring"""
+
+    mode: str
+    reason: list
+
+    @classmethod
+    def from_api_repr(cls, stats: Dict[str, Any]):
+        mode = stats.get("indexUsageMode", "INDEX_USAGE_MODE_UNSPECIFIED")
```

Comment:
Generally, we want to avoid the library injecting values not present in the API response, so returning the default value here may not be more desirable than empty/none.